packages/headless/src/database/mongodb/order/index.ts

```diff
@@ -1,32 +1,78 @@
-import { 
-  ChangeStatusEntity, 
-  GetAllOrderQueryEntity, 
-  OrderEntity, 
-  OrderIncompleteStatEntity, 
-  OrderSortQuery, 
-  OrderStatEntity, 
-  OrderStatusEnum, 
-  ShippingStatusEnum, 
-  StatusTypeDto 
+import {
+  CartItem, 
+  ReOrderQuery,
+  ChangeStatusEntity,
+  GetAllOrderQueryEntity,
+  OrderEntity,
+  OrderIncompleteStatEntity,
+  OrderSortQuery,
+  OrderStatEntity,
+  OrderStatusEnum,
+  ProductOrder,
+  ShippingStatusEnum,
+  StatusTypeDto,
 } from 'src/entity/order';
 import { IOrderDatabase } from 'src/modules/order/repositories/order.db.interface';
 import { ProductModel } from '../product/product.model';
 import { OrderModel } from './order.model';
-import { CreateOrderRequest, CreateProductOrderDetails, IProductOrderData } from 'models';
+import {
+  CreateOrderRequest,
+  CreateProductOrderDetails
+} from 'models';
+import { CartModel } from '../cart/cart.model';
 
 export class OrderDatabase implements IOrderDatabase {
+  async addToCart(userId: string, items: CartItem[]) : Promise<any> {
+    let addItems;
+    try{
+      addItems = await Promise.all(items.map(async (item) => await CartModel.findOneAndUpdate( { userId },{ $push: { items: item } },{ new: true } )));
```

Comment:
use $set operation instead of $push. here you are updating the cartmodel inside loop. and the operations are asynchoronous. one update may overwrite other update operation. also please check normal addItem api which adds items to cart. some price calculation logic should be there. keep sync with that flow.